src/wp-includes/class-wp-comment-query.php

```diff
@@ -449,7 +449,7 @@ public function get_comments() {
 		$_args = wp_array_slice_assoc( $this->query_vars, array_keys( $this->query_var_defaults ) );
 		unset( $_args['fields'], $_args['update_comment_meta_cache'], $_args['update_comment_post_cache'] );
 
-		$key          = md5( serialize( $_args ) );
+		$key          = md5( serialize( asort( $_args ) ) );
```

Comment:
For future reference, the `asort` would need to be on the line prior as it sorts by reference and returns a boolean `true`. I'd suggest `ksort` too.

Given you found this to be invalid, you can file all this information under Cool Story, Peter. :D

Reply:
I found this one myself, as tests failed. I was about to push the change I found the real issue.